sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -330,6 +330,19 @@ public void TearDownRecordedTestClass()
             }
         }
 
+        protected async Task SetProxyOptionsAsync(ProxyOptions options)
+        {
+            if (Mode == RecordedTestMode.Record && options != null)
+            {
+                await _proxy.Client.SetRecordingTransportOptionsAsync(Recording.RecordingId, options).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// If set to true, the proxy will be configured to connect on ports 5000 and 5001. This is useful when running the proxy locally for debugging recorded test issues.
+        /// </summary>
+        protected bool UseLocalDebugProxy { get; set; }
```

Comment:
We should use the same pattern as SaveDebugRecordingsOnFailure property where we throw if this is running in CI. Also, group with the other properties, ideally right below SaveDebugRecordingsOnFailure.